cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -111,12 +115,18 @@ public void setUp() throws Exception {
       doClear(configuredNamespace, false);
     }
     registeredNamespaces.put(configuredNamespace, deleteUponTeardown);
+    applicationManagers.clear();
     LOG.info("Completed setUp.");
   }
 
   @After
   public void tearDown() throws Exception {
     LOG.info("Beginning tearDown.");
+
+    // Stop any running programs created in @Test
+    for (ApplicationManager manager: applicationManagers) {
```

Comment:
Instead of modifying the `ProgramClient#stopAll` method to wait for the run record, would it be better to change the `/status` endpoint of a program to return the status based upon the run record?